resources/yki/queries.sql

```diff
@@ -110,6 +110,87 @@ INSERT INTO exam_language (
   (SELECT id FROM organizer WHERE oid = :oid AND deleted_at IS NULL)
 );
 
+-- name: delete-quarantine!
+DELETE FROM quarantine WHERE id = :id;
+
+-- name: select-quarantine
+SELECT
+  q.id,
+  q.language_code,
+  q.end_date,
+  q.birthdate,
+  q.created,
+  q.ssn,
+  q.name,
+  q.email,
+  q.phone_number
+FROM quarantine q;
+
+-- name: insert-quarantine!
+INSERT INTO quarantine (
+  language_code,
+  end_date,
+  birthdate,
+  ssn,
+  name,
+  email,
+  phone_number
+) VALUES (
+  :language_code,
+  :end_date,
+  :birthdate,
+  :ssn,
+  :name,
+  :email,
+  :phone_number
+);
+
+-- name: select-quarantine-matches
+SELECT
+  q.id,
+  q.language_code AS quarantine_lang,
+  q.end_date,
+  q.birthdate,
+  q.created,
+  q.ssn,
+  q.name,
+  q.email,
+  q.phone_number,
+  r.reviewed,
+  r.id AS registration_id,
+  r.quarantine_id,
+  r.form,
+  ed.exam_date,
+  es.language_code
+FROM quarantine q
+JOIN registration r
+-- TODO Constrain to registrations with now() <= exam_date <= q.end_date
+-- TODO Possibly filter out registrations that are reviewed?
```

Comment:
Kyllä, vertailtava kutakuinkin `r.quarantine_id IS NULL AND (r.reviewed IS NULL OR r.reviewed <= q.updated)` olettaen että q.updated on luonnin jälkeen sama kuin q.created. Tarkoitus on siis että ilmoittautumisen reviewin jälkeen luotu tai muokattu quarantine tulee uudelleen tarkistukseen ellei jo merkitty karenssiin.

Reply:
Juu, `updated` ja `created` saavat molemmat saman arvon luonnin yhteydessä.